### PR TITLE
Keshav | Week 4 - Implemented react routing & utilized API calls for data retrieval 

### PR DESCRIPTION
**Pull request by choudhary-keshav**
Pull Request Questionnaire

(1) Which Jira ticket(s) is this pull request for, and is the ticket now complete?

(2) Briefly describe your change.
I have integrated my week-4 assignment in this joint project. In my week-4 assignment I have added a weather component for knowing whether details of any city by OpenWeatherMap API, and also made a Tic-Tac-Toe game and added dynamic routing for navigating between these two components.

(3) What parts of the application are affected by your change?
It'd just affect the "Keshav's", "week-4" component.

(4) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Tested my application on various edge cases and some common scenarios

### CODE REVIEW

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx [hunk 1]
```diff
@@ -36,9 +36,10 @@ const WeatherDisplay = () => {
               description: data.weather[0].description
             })
           : null;
+        console.log(data);
       })
       .catch((error) => {
-        console.log(error);
+        console.log(error.statusCode, 'rnljsngr');
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx]: What is  `rnljsngr`

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/styles.ts [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
+    cursor: pointer;
+    padding: 10px 20px;
+    border: none;
+    border-radius: 5px;
+    font-size: 16px;
+    background-color: #ff5945;
+    color: #fff;
+    transition: background-color 0.3s ease;
+    margin-bottom: 20px;
+  }
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/styles.ts:16]: repeated css

File: interns-react-weekly-assignments/src/components/Home/Keshav/index.tsx [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
-import { Link, Outlet } from 'react-router-dom'
-import { KeshavWrapper } from './styles'
+import React, { useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
+import { KeshavWrapper } from './styles';
 
 const Keshav = () => {
-
-  const [selectedWeek, setSelectedWeek] = useState("");
+  const [selectedWeek, setSelectedWeek] = useState('');
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Keshav/index.tsx]: type

File: interns-react-weekly-assignments/src/components/Home/Keshav/index.tsx [hunk 4]
```diff
@@ -1,34 +1,41 @@
-import React, { useState } from 'react'
-import { Link, Outlet } from 'react-router-dom'
-import { KeshavWrapper } from './styles'
+import React, { useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
+import { KeshavWrapper } from './styles';
 
 const Keshav = () => {
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Keshav/index.tsx]: Type

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx [hunk 5]
```diff
@@ -36,9 +36,10 @@ const WeatherDisplay = () => {
               description: data.weather[0].description
             })
           : null;
+        console.log(data);
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx]: use console.error for errors

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx [hunk 6]
```diff
@@ -71,15 +72,11 @@ const WeatherDisplay = () => {
...
         ) : apiCode === '429' ? (
-          <p className='apiError'>Too many requests! You made more than 60 API requests in a minute!</p>
+          <p className='apiError'>{ERROR_429}</p>
         ) : apiCode === '401' ? (
-          <p>
-            You did not specify your API key in API request. (or) Your API key is not activated yet. Within the next couple of hours, it
-            will be activated and ready to use. (or) You are using wrong API key in API request. Please, check your right API key in
-            personal account. (or) You are using a Free subscription and try requesting data available in other subscriptions .
-          </p>
+          <p>{ERROR_401}</p>
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx:78]: use proper error specific names

**Review by shreegovinda — CHANGES_REQUESTED** [review]

**Review by anusreesubash — APPROVED** [review]

**Review by shreyansh-khatri — APPROVED** [review]

**Review by shreegovinda — APPROVED** [review]
